src/main/java/ch/njol/skript/expressions/ExprNewStatisticValue.java

```diff
@@ -0,0 +1,70 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.EventRestrictedSyntax;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerStatisticIncrementEvent;
+import org.jetbrains.annotations.Nullable;
+
+@Name("New/Previous Statistic Value")
+@Description("Get the new or previous value of a statistic after it has been incremented or changed.")
+@Examples({
+	"on player statistic change:",
+		"\tset {_new} to new statistic value",
+		"\tset {_old} to previous statistic value"
+})
+@Since("INSERT VERSION")
+@Events("Player Statistic Change")
+public class ExprNewStatisticValue extends SimpleExpression<Number> implements EventRestrictedSyntax {
+
+	static {
+		Skript.registerExpression(ExprNewStatisticValue.class, Number.class, ExpressionType.SIMPLE,
+			"(future|new) statistic [value]", "(past|previous) statistic [value]");
```

Comment:
This whole class should be an EventValueExpression. What you do is register the numbers as event values TIME_FUTURE and TIME_PAST then the EventValueExpression will automatically be able to grab the future and past values. This is because there is already a syntax for future and past in ExprEventValue

Reply:
I'd like to re-open this a bit because Lime is correct that a EVE would be more appropriate, since then you use setTime to control past/future automatically. However, I think that would leave the syntax as `statistic [value]` which would interfere with the classinfo itself. That said, `statistic value` is a less ambiguous name and doesn't conflict at all, which I think I would prefer to see.